src/iosource/pcap/Source.cc

```diff
@@ -263,17 +263,12 @@ void PcapSource::DoneWithPacket()
 	// Nothing to do.
 	}
 
-bool PcapSource::PrecompileFilter(int index, const std::string& filter)
-	{
-	return PktSrc::PrecompileBPFFilter(index, filter);
-	}
-
 detail::BPF_Program* PcapSource::CompileFilter(const std::string& filter)
 	{
 	std::string errbuf;
```

Comment:
This errbuf is now unused within this function?

Should the following lines from the PktSrc class be replicated within here as well?

https://github.com/zeek/zeek/pull/2396/files#diff-a7e13a32da0ee2af046a24b86a43b3325625943ee076789c70bc728575dde1d0R214-R215

I'm having a bit of a hard-time piecing this all together, but without the lines, the actual filter_error won't be propagated to the sources's `errbuf` that is then later retrieved via `Pcap::error()`, correct?

https://github.com/zeek/zeek/blob/f8bc23d3e1a6542b8e2212537200b805e49992a6/scripts/base/frameworks/packet-filter/main.zeek#L287-L292

Reply:
Yep, fixed. I missed updating that in one of the prior commits.

Reply:
> [zeek/scripts/base/frameworks/packet-filter/main.zeek](https://github.com/zeek/zeek/blob/f8bc23d3e1a6542b8e2212537200b805e49992a6/scripts/base/frameworks/packet-filter/main.zeek#L287-L292)

This part's definitely a bit of a problem. Basically this is the cause of the notice I mentioned below. `Pcap::precompile_filter` doesn't actually fail anymore except on `FATAL` problems, so it doesn't fall into this block. `install_pcap_filter` *does* fail though, so it reports the notice later on. I can confirm that does the right thing on `FATAL` errors though:

```
tim@coregeek2 build% zeek -f "kaput" -r ~/Downloads/pop3.pcap
error: Error returned during BPF configuration: pcap_compile(kaput): can't parse filter expression: syntax error
fatal error in /Users/tim/Desktop/projects/zeek/scripts/base/frameworks/packet-filter/./main.zeek, line 292: cannot compile BPF filter "kaput": pcap_compile(kaput): can't parse filter expression: syntax error
```

I'll have to look into why `install_pcap_filter` doesn't like what's happening.

Reply:
Alright, `install_pcap_filter` is failing because although `precompile_filter` didn't return a failure it also didn't store a `BPF_Filter` object into the `DefaultPcapFilter` index. That's because `BPF_Filter` didn't really succeed at compiling the filter. I wonder if it would be better for the `BPF_Filter` object to exist but store the compilation state locally instead of returning it back to the PktSrc. I'll have to consider that one a bit more.

This is definitely a want-to-have change, so it's ok if it doesn't make it into 5.1. I'd rather get it right first than rush it through.